### PR TITLE
cjson: add utils option

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **cjson/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/3559

### CODE REVIEW

File: recipes/cjson/all/conanfile.py [hunk 1]
```diff
@@ -15,11 +15,13 @@ class CjsonConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "cjson_utils": [True, False],
```
Comment (madebr) [inline recipes/cjson/all/conanfile.py:18]: Would `"utils"` be fine too?

Comment (SpaceIm) [reply recipes/cjson/all/conanfile.py:18]: I guess.
In the meantime cjson_utils is closer to upstream option name: https://github.com/DaveGamble/cJSON/blob/4100379a04f2f2838ffb00c4ad3d0dd9e49f4147/CMakeLists.txt#L172

Comment (madebr) [reply recipes/cjson/all/conanfile.py:18]: Yes, but they use the `CJSON_` prefix for everyting:
https://github.com/DaveGamble/cJSON/blob/4100379a04f2f2838ffb00c4ad3d0dd9e49f4147/CMakeLists.txt#L232

Comment (SpaceIm) [reply recipes/cjson/all/conanfile.py:18]: Actually only for these 2 options, none of others are prefixed. But we can rename this option to `utils`, it's understandable for consumers.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/cjson/all/conanfile.py [hunk 2]
```diff
@@ -44,6 +45,8 @@ def configure(self):
             del self.options.fPIC
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
+        if self.settings.compiler == "Visual Studio" and self.options.shared and self.settings.compiler.runtime == "MTd":
```
Comment (madebr) [inline recipes/cjson/all/conanfile.py:48]: I think you can skip MT too.
If MTd fails hard, then MT will fail subtly.

Comment (SpaceIm) [reply recipes/cjson/all/conanfile.py:48]: MT doesn't fail

Comment (madebr) [reply recipes/cjson/all/conanfile.py:48]: The main difference between MT and MTd are the debug assertions. The fact that MTd asserts, means that MT will fail too.

Otoh, libffi MTd fails too and we keep MT available.

**Review by theirix — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Pull request merged**